subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportViewerTestReporter.kt

```diff
@@ -60,84 +64,108 @@ internal class ReportViewerTestReporter(
+                            )
+
+                            // send only for failed tests
+                            val (stdout: String, stderr: String) = if (testRuntimeData.incident != null) {
+                                logcatBuffers.getLogcat(key)
+                            } else {
+                                "" to ""
+                            }
+
+                            report.sendCompletedTest(
+                                AndroidTest.Completed.create(
+                                    testStaticData = testFromSuite,
+                                    testRuntimeData = testRuntimeData,
+                                    stdout = stdout,
+                                    stderr = stderr
+                                )
+                            )
+                        } catch (throwable: Throwable) {
+                            val (stdout: String, stderr: String) = logcatBuffers.getLogcat(key)
+
+                            val errorMessage = "Can't parse testRuntimeData: ${test.testName}; ${reportJson.readText()}"
 
-                    report.sendCompletedTest(
-                        AndroidTest.Completed.create(
-                            testStaticData = testFromSuite,
-                            testRuntimeData = testRuntimeData,
+                            logger.warn(errorMessage, throwable)
+
+                            sendLostTest(
+                                testFromSuite = testFromSuite,
+                                errorMessage = errorMessage,
+                                stdout = stdout,
+                                stderr = stderr,
+                                throwable = throwable
+                            )
+
+                            metricsSender.sendReportFileParseErrors()
+                        }
+                    },
+                    { throwable ->
+                        val (stdout: String, stderr: String) = logcatBuffers.getLogcat(key)
+
+                        val errorMessage = "Can't get report from file: $test"
+
+                        logger.warn(errorMessage, throwable)
+
+                        sendLostTest(
+                            testFromSuite = testFromSuite,
+                            errorMessage = errorMessage,
                             stdout = stdout,
-                            stderr = stderr
+                            stderr = stderr,
+                            throwable = throwable
                         )
-                    )
-                } catch (throwable: Throwable) {
-                    val (stdout: String, stderr: String) = logcatBuffers.getLogcat(key)
 
-                    val errorMessage = "Can't parse testRuntimeData: ${test.testName}; ${reportJson.readText()}"
+                        metricsSender.sendReportFileNotAvailable()
+                    }
+                )
+            }
+            is TestResult.InCompleted -> {
```

Comment:
Before we tried to pull artifacts from the device even when test execution was `incomplete` and it led us to replace the original error.
Now we understand that test was `incomplete` and we don't try to pull artifacts from the device